### PR TITLE
Email Verification

### PR DESCRIPTION
**Pull request by aivantg**
[//]: # "These comments meant for your reference, they are invisible and don't need to be deleted"

## What's new in this PR?

[//]: # "Describe what's new in this PR"

- Updated to latest version of Airlock
- Added back Owner Access Resolvers
- Added endpoint for email uniqueness check
- Updated Schema Generator
- Removed endpoint URL and added second Base ID env variable for schema generator
- Removed redundant env variable verification (`dotenv.config()` does the same thing, it makes sure all variables in `.env.example` are set)

Note: I've already deployed this branch to heroku for testing

### How to Review

[//]: # 'The order in which to review files and what to expect when testing locally'

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc"

Corresponding pp-web PR: https://github.com/calblueprint/peoplepower-web/pull/181

### Next steps

[//]: # "What doesn't work yet, what's NOT in this PR/has to be done "

### Tests Performed, Edge Cases

[//]: # 'If you made changes/additions to `airtable.js` make sure you update/add'
[//]: # '[at least 1] corresponding unit tests in `airtable.spec.js`'

### Screenshots

[//]: # "Add screenshots!!! If you'd like"

CC: @dfangshuo

### CODE REVIEW

File: airtable/airtable.js [hunk 1]
```diff
@@ -10,20 +10,21 @@
...
-require('dotenv-safe').config();
+const API_KEY = process.env.REACT_APP_AIRTABLE_API_KEY;
+const ENDPOINT_URL = 'https://api.airtable.com';
+const VIEW = 'Grid view';
 
 Airtable.configure({
-  endpointUrl: Constants.ENDPOINT_URL,
-  apiKey
+  endpointUrl: ENDPOINT_URL,
+  apiKey: API_KEY
```
Comment (dfangshuo) [inline airtable/airtable.js:24]: nit: i had opted for a string in the past because i thought that made it easier to see at first glance that we are using Airlock (with Airtable-esque API), but whatever works

File: airtable/request.js [hunk 2]
```diff
@@ -86,6 +86,15 @@ export const getAllOwners = async (filterByFormula = '', sort = []) => {
   return getAllRecords(Tables.Owner, filterByFormula, sort);
 };
 
+export const getOwnersByEmail = async (value, sort = []) => {
```
Comment (dfangshuo) [inline airtable/request.js:89]: might be a good idea to post-fix this with unsafe e.g. getOwnersByEmailUnsafe or otherwise indicate that this is a potentially dangerous function... 

Also some comments about why  we need to do this (limitations of airlock) might be helpful for the future too

File: index.js [hunk 3]
```diff
@@ -11,36 +11,17 @@ import generatePdfForSubscriber from './utils/pdfgeneration';
...
-
-// Fails explicitly if the environment is improperly configured
-AppConfigurations.forEach(param => {
-  if (!process.env[param]) {
-    throw new Error(
-      `Required configuration variable ${param} is ${process.env[param]}. Do you have a .env file and is it setup correctly?`
-    );
-  }
-});
-
```
Comment (dfangshuo) [inline index.js:40]: is there a reason you're opposed to this? I personally think it's a good idea for the app to fail explicitly, so we know from the get-go why  it hasn't been properly configured, as opposed to it throwing up silent errors later on

**Review by dfangshuo — APPROVED** [review]
1 question about failing explicitly,  but otherwise looks good to me!

**Comment by dfangshuo** [comment]
To add  on, the node server currently uses 2 API keys, 1 is an actual airtable API key in `airtable.js`, the other is the string 'airlock'. In both of these cases, we are passing them into a constructor called 'Airtable(...)'. I think it's good to make this distinction clear through  code to  avoid further confusion

File: airtable/airtable.js (hunk 1 above)
Comment (aivantg) [reply airtable/airtable.js:24]: yeah i mean for the front end this is the string “airlock” but the backend is not using airlock so it needs an actual API KEY

File: airtable/request.js (hunk 2 above)
Comment (aivantg) [reply airtable/request.js:89]: why is this potentially unsafe? This is on the backend so it doesn’t go through airlock

File: index.js (hunk 3 above)
Comment (aivantg) [reply index.js:40]: the app DOES fail explicitly. The package I used to load environment variables `dotenv-safe` makes sure that all the variables that are in .env.example are loaded. Otherwise the app won’t start.

Comment (aivantg) [reply index.js:40]: not opposed at all, we just already have it :)

**Comment by aivantg** [comment]
Hm, I don’t remember there being two API keys? The node server should never use the API Key of “airlock”, it is a “super admin” of sorts and will edit airtable directly, so it needs its own API key and uses the original airtable package

**Comment by dfangshuo** [comment]
you're right! I realized my mistake (re: the API keys)

File: airtable/request.js (hunk 2 above)
Comment (dfangshuo) [reply airtable/request.js:89]: unsafe as in to denote that this function can potentially return the data of every single person in the database, but also I'm not hung up on this
